File: C#/LocalSearchOptimization/Parameters/OptimizationParameters.cs
namespace LocalSearchOptimization.Parameters
{
    public class OptimizationParameters
    {
        public string Name { get; set; } = "core";

        public int Seed { get; set; } = 0;

        public bool DetailedOutput { get; set; } = false;

        public OptimizationParameters()
        {

        }

        protected OptimizationParameters(OptimizationParameters copy)
        {
            Name = copy.Name;
            Seed = copy.Seed;
            DetailedOutput = copy.DetailedOutput;
        }

        public virtual OptimizationParameters Clone()
        {
            return new OptimizationParameters(this);
        }
    }
}

File: java/src/localsearchoptimization/parameters/OptimizationParameters.java
package localsearchoptimization.parameters;

public class OptimizationParameters {

    public String name = "core";

    public int seed = 0;

    public boolean isDetailedOutput = true;

    public OptimizationParameters() {

    }

    protected OptimizationParameters(OptimizationParameters copy) {
        name = copy.name;
        seed = copy.seed;
        isDetailedOutput = copy.isDetailedOutput;
    }

    public OptimizationParameters Clone() {
        return new OptimizationParameters(this);
    }
}